crates/storage-ops/src/rollback/node/batch_prover.rs

```diff
@@ -83,26 +86,58 @@ impl BatchProverLedgerRollback {
 
             batch.delete::<ProverPendingCommitments>(&comm_idx)?;
             increment_table_counter!("ProverPendingCommitments", rollback_result);
+        }
 
-            let mut jobs_iter = self
-                .ledger_db
-                .iter_with_direction::<CommitmentIndicesByJobId>(
-                    Default::default(),
-                    ScanDirection::Backward,
-                )?;
-            jobs_iter.seek_to_last();
-            for job_record in jobs_iter {
-                let job_record = job_record?;
-                let job_id = job_record.key;
-                let job_commitment_indices = job_record.value;
-                if !job_commitment_indices.contains(&comm_idx) {
-                    continue;
-                }
-                batch.delete::<CommitmentIndicesByJobId>(&job_id)?;
-                increment_table_counter!("SequencerCommitmentByIndex", rollback_result);
+        // Now handle jobs that might contain commitments we're rolling back
+        let mut jobs_to_check = Vec::new();
+        let mut jobs_iter = self
+            .ledger_db
+            .iter_with_direction::<CommitmentIndicesByJobId>(
+                Default::default(),
+                ScanDirection::Backward,
+            )?;
+        jobs_iter.seek_to_last();
+
+        // Collect jobs that have any indices above our rollback point
+        for job_record in jobs_iter {
+            let job_record = job_record?;
+            let job_id = job_record.key;
+            let commitment_indices = job_record.value;
+
+            if commitment_indices
+                .iter()
+                .any(|&idx| idx > last_sequencer_commitment_index)
```

Comment:
If we replace any with all to make sure that all indices belonging to a job are larger than the last sequencer commitment index.